File: components/physics/src/main/java/org/storm/physics/entity/SimpleEntity.java
package org.storm.physics.entity;

import lombok.NonNull;
import org.storm.physics.math.geometry.shapes.Shape;

public class SimpleEntity extends Entity {

  public SimpleEntity(@NonNull Shape hurtBox, double speed, double mass, double restitution) {
    super(hurtBox, speed, mass, restitution);
  }

}

File: components/physics/src/main/kotlin/org/storm/physics/entity/SimpleEntity.kt
package org.storm.physics.entity

import org.storm.physics.math.geometry.shapes.Shape

class SimpleEntity(boundaries: MutableMap<String, Shape>, speed: Double, mass: Double, restitution: Double) :
    Entity(boundaries, speed, mass, restitution) {

    constructor(
        boundary: Shape,
        speed: Double,
        mass: Double,
        restitution: Double
    ) : this(mutableMapOf(SINGLE_BOUNDARY to boundary), speed, mass, restitution)

}
